src/components/monitor/HeroSection.tsx

```diff
@@ -1,47 +1,69 @@
-import { useState, useEffect } from 'react';
+import React, { useState } from 'react';
+import axios from 'axios';
 import Header from '@/components/shared/Header';
 import EmailToolbar from './components/EmailToolBar';
-import { HaveIbeenPwnedText } from '@/assets/types/have-i-been-pawned';
-import { PwnedSection } from './PwnedSection';
-import { AllGoodSection } from './AllGoodSection';
+import { HaveIbeenPwnedText, Breach, Paste } from '@/assets/types/have-i-been-pawned';
+import PwnedStatusSection from './PwnedStatusSection';
+import LoadingPulse from '../shared/loader/LoadingPulse';
 
 interface HeroSectionProps {
   textContent: HaveIbeenPwnedText['HeroSection'];
 }
 
+type ViewProps = 'default' | 'success' | 'loading' | 'error';
+
 export const HeroSection: React.FC<HeroSectionProps> = ({ textContent }) => {
-  const [result, setResult] = useState<any[]>([]);
-  const [resultPastes, setResultPastes] = useState<any[]>([]);
-  const [error, setError] = useState<string | null>(null);
-  const [queryMade, setQueryMade] = useState(false);
-  const [showResult, setShowResult] = useState(false);
+  const [breaches, setBreaches] = useState<Breach[]>([]);
+  const [resultPastes, setResultPastes] = useState<Paste[]>([]);
```

Comment:
nit: maybe it would be better to rename this to `pastes` directly, what do you think? @jaaaaavier

Reply:
Yeah, much better, Updated. Also I will change onResultChange to onResultBreachesChange for be clear at all